Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt

```diff
@@ -0,0 +1,78 @@
+package com.example.reply.ui
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.PermanentNavigationDrawer
+import androidx.compose.material3.ModalNavigationDrawer
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Modifier
+import kotlinx.coroutines.flow.StateFlow
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ReplyApp(
+    windowSize: WindowSize,
+    devicePosture: StateFlow<DevicePosture>
+) {
+    val foldingDevicePosture = devicePosture.collectAsState().value
+
+    /**
+     * This will help us select type of navigation and content type depending on window size and
+     * fold state of the device.
+     *
+     * In the state of folding device If it's half fold in BookPosture we want to avoid content
+     * at the crease/hinge
+     */
+    val (navigationType, contentType) = when (windowSize) {
+        WindowSize.Compact -> {
+            Pair(ReplyNavigationType.BOTTOM_NAVIGATION, ReplyContentType.LIST_ONLY)
+        }
+        WindowSize.Medium -> {
+            if (foldingDevicePosture != DevicePosture.NormalPosture) {
+                Pair(ReplyNavigationType.NAVIGATION_RAIL, ReplyContentType.LIST_AND_DETAIL)
+            } else {
+                Pair(ReplyNavigationType.NAVIGATION_RAIL, ReplyContentType.LIST_ONLY)
+            }
+        }
+        WindowSize.Expanded -> {
+            if (foldingDevicePosture is DevicePosture.BookPosture) {
+                Pair(ReplyNavigationType.NAVIGATION_RAIL, ReplyContentType.LIST_AND_DETAIL)
+            } else {
+                Pair(ReplyNavigationType.FIXED_NAVIGATION_DRAWER, ReplyContentType.LIST_AND_DETAIL)
+            }
+        }
+    }
+
+    if (navigationType == ReplyNavigationType.FIXED_NAVIGATION_DRAWER) {
+        PermanentNavigationDrawer(drawerContent = {}) {
+            ReplyAppContent(navigationType, contentType)
+        }
+    } else {
+        ModalNavigationDrawer(drawerContent = {}) {
```

Comment:
Hey yes you are right when Bottom Navigation is active it will not show navigation drawer. With Navigation Rail it will be shown so for simplicity I will keep Drawer as main top wrapper and shown or not will be decided on dynamic states of fold and size.

Reply:
Ah gotcha, so this will be handled in upcoming code to managed the drawer state